### ISSUE TITLE
is there a way to use my custom song parser? 

### ISSUE THREAD

**Issue by mfuentesg** [martijnversluis/ChordSheetJS#800]
I was checking the codebase, but I'm not sure if it is possible to pass a custom parser but, without duplicating the whole logic that you already have for TextFormatter or HtmlDivFormatter.
I want to have just the body of the song without all the metadata.

Any clue?

**Comment by martijnversluis** [comment]
Hey @mfuentesg,

Am I correct you want to customise the text/HTML output of a song? A parser parses the document into a `Song`, and a formatter can turn it into text or HTML. 

To customise output you can either:

- write your own formatter, like [`TextFormatter`](https://github.com/martijnversluis/ChordSheetJS/blob/master/src/formatter/text_formatter.ts)
- inherit from HtmlFormatter and supply your own template, like [`HtmlDivFormatter`](https://github.com/martijnversluis/ChordSheetJS/blob/master/src/formatter/html_div_formatter.ts) ([this is the template file](https://github.com/martijnversluis/ChordSheetJS/blob/master/src/formatter/templates/html_div_formatter.ts))

Please let me know if that's helpful.

**Comment by mfuentesg** [comment]
Hey HtmlFormatter is not exported right?

**Comment by martijnversluis** [comment]
You're completely right. I'm gonna release a new version this week which exports `HtmlFormatter` (and maybe other useful classes)

**Comment by martijnversluis** [comment]
@mfuentesg I released `7.15.0` that includes exports for `Formatter` and `HtmlFormatter` 👍

**Comment by mfuentesg** [comment]
Cool! I'll check it later :)
